handlers/index.js

```diff
@@ -0,0 +1,199 @@
+            session = new Session(sid);
+        }
+
+        let offlineTokenRes =
+            await platformConfig.oauthClient.getOfflineAccessToken(
+                ext.scopes,
+                code
+            );
+
+        session.company_id = companyId;
+        session.scope = ext.scopes;
+        session.state = fdkSession.state;
+        session.extension_id = ext.api_key;
+        offlineTokenRes.access_token_validity =
+            platformConfig.oauthClient.token_expires_at;
+        offlineTokenRes.access_mode = "offline";
+        session.updateToken(offlineTokenRes);
+        await SessionStorage.saveSession(session);
+    }
+    
+    if (
+        ext.webhookRegistry.isInitialized &&
+        ext.webhookRegistry.isSubscribeOnInstall
+    ) {
+        const client = await ext.getPlatformClient(companyId, fdkSession);
+        await ext.webhookRegistry.syncEvents(client, null, true).catch((err) => {
+            logger.error(err);
+        });
+    }
+
+    let redirectUrl = await ext.callbacks.auth(reqObj);
+    logger.debug(`Redirecting after auth callback to url: ${redirectUrl}`);
+    return {
+        redirectUrl: redirectUrl,
+        fdkSession: fdkSession,
+    };
+};
+
+const fpAutoInstall = async (reqObj, company_id, code, ext) => {
+    logger.debug(
+        `Extension auto install started for company: ${company_id} on company creation.`
+    );
+
+    let platformConfig = ext.getPlatformConfig(company_id);
+    let sid = Session.generateSessionId(false, {
+        cluster: ext.cluster,
+        companyId: company_id,
+    });
+
+    let session = await SessionStorage.getSession(sid);
+    if (!session) {
+        session = new Session(sid);
+    } else if (session.extension_id !== ext.api_key) {
+        session = new Session(sid);
+    }
+
+    let offlineTokenRes = await platformConfig.oauthClient.getOfflineAccessToken(
+        ext.scopes,
+        code
+    );
+
+    session.company_id = company_id;
+    session.scope = ext.scopes;
+    session.state = uuidv4();
+    session.extension_id = ext.api_key;
+    offlineTokenRes.access_token_validity =
+        platformConfig.oauthClient.token_expires_at;
+    offlineTokenRes.access_mode = "offline";
+    session.updateToken(offlineTokenRes);
+
+    if (!ext.isOnlineAccessMode()) {
+        await SessionStorage.saveSession(session);
+    }
+
+    if (
+        ext.webhookRegistry.isInitialized &&
+        ext.webhookRegistry.isSubscribeOnInstall
+    ) {
+        const client = await ext.getPlatformClient(company_id, session);
+        await ext.webhookRegistry.syncEvents(client, null, true).catch((err) => {
+            logger.error(err);
+        });
+    }
+    logger.debug(
+        `Extension installed for company: ${company_id} on company creation.`
+    );
+    if (ext.callbacks.auto_install) {
+        await ext.callbacks.auto_install(reqObj);
+    }
+    return;
+};
+
+const fpUninstall = async (reqObj,company_id, ext) => {
```

Comment:
Add named function here as it will be helpful while reading stack trace.  @jinalvirani

Reply:
Resolved